modules/system/models/MailLayout.php

```diff
@@ -68,13 +68,18 @@ public static function listCodes()
 
     public static function getIdFromCode($code)
     {
-        $layoutId = array_get(self::listCodes(), $code);
-        if ($layoutId === null) {
-            self::createLayouts();
-            self::$codeCache = null;
-            $layoutId = array_get(self::listCodes(), $code);
+        return array_get(self::listCodes(), $code);
+    }
+
+    public static function findOrMakeLayout($code)
+    {
+        $layout = self::whereCode($code)->first();
+        if (!$layout && View::exists($code)) {
+            $layout = new self;
+            $layout->code = $code;
+            $layout->fillFromView($code);
         }
-        return $layoutId;
+        return $layout;
     }
```

Comment:
This is a great solution, although a couple minor changes required. First, we'll have to retain the `getIdFromCode()` method as it's public and could be relied on by current plugins, so just add the `findOrMakeLayout()` as an additional method. 

Secondly, could you add an extra line of whitespace before and after the `if (!$layout && View::exists($code))` block to match the `findOrMakeTemplate()` method on the MailTemplate class?

Thanks for your work on this!

Reply:
I added the space lines. 
But I can not change the method getIdFromCode to protected because it is used in `MailTemplate::syncAll()` method

Reply:
That's fine, I wanted it to stay public